test/tools/reporter/mongodb_reporter.js

```diff
@@ -229,10 +229,15 @@ class MongoDBMochaReporter extends mocha.reporters.Spec {
   }
 
   /**
-   * @param {MongoMochaTest & {skipReason?: string}} test
+   * @param {MongoMochaTest} test
    */
   pending(test) {
     if (REPORT_TO_STDIO) console.log(chalk.cyan(`↬ ${test.fullTitle()}`));
+    if (test.metadata && test.metadata.skipReason && typeof test.metadata.skipReason === 'string') {
```

Comment:
I added this specifically for the deadlock tests, which all share a metadata object but all needed to be skipped.  Rather than adding the skipReason for each test individually, I skipped the whole suite by attaching it to the metadata. 

I actually prefer the metadata approach because it keeps all data related to why a test may be skipped together but don't feel strongly about this.  If you'd prefer to have a single method for attaching skip reasons to tests, I can revert this change

Reply:
I think having multiple ways to use a feature makes that feature more brittle (i.e., higher chance of breaking without noticing in one place or the other), so my personal preference would be to avoid it; but I appreciate the convenience and the visibility afforded by having it in metadata, so if you don't find the risk of the brittleness compelling here, I'm fine with leaving it as is

Reply:
What about keeping the `skipReason` made by the metadata filters and appending additional context. "skipped because of an auth disabled requirement" is still useful to see, but then we could have a `skipContext` or `metaSkipReason` (or maybe keep it simple `note: 'xxx'`) that if defined appends more to the message so that the logs reflect both, skipped cus auth and cus of a TODO.

Reply:
The funny thing about this logic is that it's going to append that reason right now no matter what the cause of the skip is, so, for example, when they don't run on single topologies, we'd still get the same metadata skipReason, if I am reading this correctly, since the skipReason isn't attached to a specific filter

Reply:
Okay I think that's a valid point and that could lead to confusion, we have the skipReason telling us the reason was auth and we'll keep the TODO to track the unskipping for the particular environment with no auth enabled. I'm okay with that.